packages/messaging/src/helpers/idb-manager.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DB, openDb, deleteDb } from 'idb';
+import { TokenDetails } from '../interfaces/token-details';
+import { migrateOldDatabase } from './migrate-old-database';
+import { FirebaseInternalDependencies } from '../interfaces/internal-dependencies';
+
+// Exported for tests.
+export const DATABASE_NAME = 'firebase-messaging-database';
+const DATABASE_VERSION = 1;
+const OBJECT_STORE_NAME = 'firebase-messaging-store';
+
+let dbPromise: Promise<DB> | null = null;
+function getDbPromise(): Promise<DB> {
+  if (!dbPromise) {
+    dbPromise = openDb(DATABASE_NAME, DATABASE_VERSION, upgradeDB => {
+      // We don't use 'break' in this switch statement, the fall-through
```

Comment:
Solid logic, if a little confusing when you only have one case :smile:

Reply:
I think it's still helpful to whoever's going to maintain this, telling them that in case they want to add a mutation, this is a good way to do so.

And definitely much better than having this comment, and actual break statements right afterwards. :sweat_smile:

https://github.com/firebase/firebase-js-sdk/blob/7250d25145f27ee099dc717a0130c2f8dc9841d2/packages/messaging/src/models/token-details-model.ts#L101-L102